File: public/examples/go/basic/02_variables.go
package main

import "fmt"

func main() {
    // Standard declaration
    var name string = "Alice"
    var age int = 25
    
    // Short declaration (inside functions only)
    pi := 3.14
    isHappy := true
    
    fmt.Printf("Name: %s, Age: %d, Pi: %f, Happy: %t\n", name, age, pi, isHappy)
    
    // Zero values
    var x int
    fmt.Println("Zero value of int:", x)
}

File: examples/cpp/basic/02_variables.cpp
#include <iostream>
#include <string>

int main() {
    // Standard data types
    int age = 25;
    double pi = 3.14159;
    std::string name = "Alice";
    bool isHappy = true;
    
    // Type inference (C++11+)
    auto status = "Learning";
    
    std::cout << "Name: " << name << ", Age: " << age << std::endl;
    std::cout << "Status: " << status << std::endl;
    
    return 0;
}
